core/src/apps/cardano/address.py

```diff
@@ -34,6 +40,45 @@
 MAX_ADDRESS_BYTES_LENGTH = 65
 
 
+def validate_address_parameters(parameters: CardanoAddressParametersType) -> None:
+    if parameters.address_type == CardanoAddressType.BYRON:
```

Comment:
to rule out any ambiguity, I'd enforce that the rest of fields of parameters should not be set, something like
```python
if any([parameters.address_n_staking, parameters.staking_key_hash]):
    raise INVALID_ADDRESS_PARAMETERS
```

to make it less cumbersome, the individual parameters may be extracted into separate variables so you don't have to prefix it with "params." all the time

Reply:
What do you think of this - https://github.com/vacuumlabs/trezor-firmware/pull/60/commits/b09e312a9aa6e275b8ef1430822364757ce15cd4?

Reply:
sorry for the delayed reply - looks good to me :)